### PR TITLE
Reader: Related Posts

### PR DESCRIPTION
**Pull request by blowery**
Add actions and reducer, hook up to new card style in full post view.

Things to do:
- [ ] ~~Add stats tracking to post clicks~~ DEFER
- [ ] ~~Add railcar info for stream items?~~ DEFER, needs consensus on #5658 
- [x] Hide behind a feature flag?

Since we have the feature flag now, I'd like to land this and defer stat tracking and railcars.

### CODE REVIEW

**Comment by blowery** [comment]
Needs #5564

**Comment by jancavan** [comment]
@blowery made some minor updates here: https://github.com/Automattic/wp-calypso/pull/5669/commits/8bd4eeee39fb33eebd7c1a0050accb3d652eb9a0

Design looks good to me 👍

**Comment by blowery** [comment]
@jancavan do we want a placeholder while the posts load? if we have a placeholder, do we need an "empty" variant?

**Comment by blowery** [comment]
Since we have the feature flag now, I'd like to land this and defer stat tracking and railcars.

**Comment by blowery** [comment]
Hrm, this probably needs tests too.

File: client/reader/related-posts/index.jsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+/**
+ * Internal Dependencies
+ */
+import { relatedPostsForPost } from 'state/reader/related/selectors';
+import { getPost } from 'state/reader/posts/selectors';
+import { getSite } from 'state/reader/sites/selectors';
+import SmallPost from 'components/post-card/small';
+import QueryReaderRelatedPosts from 'components/data/query-reader-related-posts';
+import i18n from 'lib/mixins/i18n';
```
Comment (Tug) [inline client/reader/related-posts/index.jsx]: Hi there, i18n usage has changed recently. please use:
```
import i18n from 'i18n-calypso';
```
instead.

File: client/components/data/query-reader-related-posts/index.jsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+ * External dependencies
+ */
+import { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import { shouldFetchRelated } from 'state/reader/related/selectors';
```
Comment (bluefuton) [inline client/components/data/query-reader-related-posts/index.jsx]: Should this part of the Redux tree be named `state/reader/related-posts` to be clear?

**Comment by bluefuton** [comment]
Worth adding tests for actions, reducer and selectors I think.

When you hit '<- back' from a suggested site or post, it'd be nice if you were anchored to the 'suggested reading' on the full post you started from, rather than the top of the post.

I think `client/reader/related-posts/index.jsx` should be in `client/reader/related-posts/main.jsx` (based on feedback @mtias recently gave us on `read/start`).

Works well though, and the suggestions I've seen have been excellent :)

File: client/components/data/query-reader-related-posts/index.jsx (hunk 2 above)
Comment (blowery) [inline client/components/data/query-reader-related-posts/index.jsx]: yeah, good call

**Comment by blowery** [comment]
@bluefuton tests added

**Comment by blowery** [comment]
moved related posts to components to alleviate the index / main issue

**Comment by blowery** [comment]
>   When you hit '<- back' from a suggested site or post, it'd be nice if you were anchored to the 'suggested reading' on the full post you started from, rather than the top of the post.

Yeah, would be nice. Probably not for the initial release though.

File: client/state/reader/reducer.js [hunk 3]
```diff
@@ -11,11 +11,13 @@ import feeds from './feeds/reducer';
 import sites from './sites/reducer';
 import start from './start/reducer';
 import posts from './posts/reducer';
+import related_posts from './related-posts/reducer';
```
Comment (bluefuton) [inline client/state/reader/reducer.js]: Should `related_posts` be camelcase?

e.g. in sites reducer:

```export default combineReducers( {
	domains,
	fetchingItems,
	items,
	mediaStorage,
	plans,
	vouchers
} );```

**Comment by bluefuton** [comment]
One tiny naming nit, and could do with a squash.

Otherwise looks ready to go!

File: client/state/reader/reducer.js (hunk 3 above)
Comment (blowery) [inline client/state/reader/reducer.js]: hmmm i thought we were keeping it in line with the api responses. i guess not. will fix.

**Pull request merged**